Minecraft/Bot.ts

```diff
@@ -0,0 +1,48 @@
+import { createClient } from '../Utils/MinecraftClient';
+import { DiscordClient } from '../Utils/DiscordClient';
+import { handleChat } from './Chat';
+import { PlayerManager } from './PlayerManager';
+
+export function handleMinecraft(client: DiscordClient) {
+    if (!client.config['in-game-bot'].enabled) return; // Returns if there is no bot config
+    client.bot = createClient({
+        host: client.config.ips.n00b,
+        port: 25565,
+        username: 'sysctl',
+        profilesFolder: __dirname + '\\auth',
+        auth: 'microsoft',
+        disableChatSigning: true
+    }); // Creates the bot
+
+    client.bot.playerManager = new PlayerManager(client.bot);
+
+    client.bot.on('disconnect', (packet) => {
+        console.log('Disconnected from server : ' + packet.reason);
+        reconnectToServer(client);
+    }); // Sends log message when the bot disconnects from the server and reconnects it
+
+    client.bot.on('end', () => {
+        console.log('Connection lost');
+        reconnectToServer(client);
+    }); // Sends log message when the bot loses connection to the server and reconnects it
+
+    client.bot.on('error', (err) => {
+        if (err.message.includes('play.toClient')) return;
+        console.log('Error occurred');
+        console.log(err);
+    }); // Sends log message when the bot encounters an error
+
+    client.bot.on('connect', () => {
+        console.log(`Connected to Server...`);
+        handleChat(client);
+    }); // Sends log message when the bot connects to the server
+}
+
+// Waits 3 seconds and tries to reconnect to the server
+function reconnectToServer(client: DiscordClient) {
```

Comment:
Spigot servers ratelimit you (connection throttled) for 5 seconds if you connect to fast so this will never work. I'd recommend setting this to 1 minute or so just to be sure (this way you won't get flagged by external firewalls either). 1 Minute of bot downtime isn't that much. Possibly makes this a config option?